tests/testthat/test-tspa.R

```diff
@@ -456,3 +456,102 @@ test_that("Multiple-group multiple-factor example", code = {
   expect_equal(sct$se[sct$op == "~"], expected = scs$se[scs$op == "~"],
                tolerance = 0.0001)
 })
+
+########## Error messages ##########
+
+test_that("Empty path model", {
+  expect_error(
+    tspa(model = "",
+         data = fs_dat_3fac,
+         fsT = attr(fs_dat_3fac, "fsT")),
+    "Please provide a structural path model."
+  )
+})
+
+test_that("Need to provide none or both fsT and fsL", {
+  expect_error(
+    tspa(model = path_mod,
+         data = fs_dat_3fac,
+         fsT = attr(fs_dat_3fac, "fsT")),
+    "Please provide both or none of fsT and fsL"
+  )
+  expect_error(
+    tspa(model = path_mod,
+         data = fs_dat_3fac,
+         fsL = attr(fs_dat_3fac, "fsL")),
+    "Please provide both or none of fsT and fsL"
+  )
+  expect_no_error(
+    tspa(model = path_mod,
+         data = fs_dat_3fac,
+         fsT = attr(fs_dat_3fac, "fsT"),
+         fsL = attr(fs_dat_3fac, "fsL"))
+  )
+})
+
+test_that("Names of factor score variables need to match those in the input data", {
+  data("PoliticalDemocracy", package = "lavaan")
+  mod2 <- "
+  # latent variables
+    ind60 =~ x1 + x2 + x3
+    dem60 =~ y1 + y2 + y3 + y4
+    dem65 =~ y5 + y6 + y7 + y8
+  "
+  fs_dat2 <- get_fs(PoliticalDemocracy, model = mod2, std.lv = TRUE)
+  ecov_fs <- attr(fs_dat2, "fsT")
+  dimnames(ecov_fs) <- lapply(dimnames(ecov_fs),
+                              FUN = \(x) paste0("bs_", x))
+  expect_error(
+    tspa(model = "dem60 ~ ind60
+              dem65 ~ ind60 + dem60",
+         data = fs_dat2,
+         fsT = ecov_fs,
+         fsL = attr(fs_dat2, "fsL")),
+    "Names of factor score variables do not match those in the input data."
+  )
+  expect_no_error(
+    tspa(model = "dem60 ~ ind60
+              dem65 ~ ind60 + dem60",
+         data = fs_dat2,
+         fsT = attr(fs_dat2, "fsT"),
+         fsL = attr(fs_dat2, "fsL"))
+  )
+})
+
+test_that("Test indicator names not starting with 'fs_'", {
```

Comment:
Nice test. I would suggest also adding a quick one that using `fs_` or `bs_` give identical results.